src/Microsoft.AspNetCore.Mvc.Razor.Precompilation.Tools/Internal/PrecompileDispatchCommand.cs

```diff
@@ -0,0 +1,239 @@
+using Microsoft.Extensions.Internal;
+using NuGet.Frameworks;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Precompilation.Internal
+{
+    public class PrecompileDispatchCommand
+    {
+        private CommonOptions Options { get; } = new CommonOptions();
+
+        private CommandOption NoBuildOption { get; set; }
+
+        public CommandOption FrameworkOption { get; set; }
+
+        public CommandOption ConfigurationOption { get; set; }
+
+        private NuGetFramework TargetFramework { get; set; }
+
+        private ProjectContext RuntimeContext { get; set; }
+
+
+        private string OutputPath { get; set; }
+
+        private string ProjectPath { get; set; }
+
+        private string Configuration { get; set; }
+
+        public void Configure(CommandLineApplication app)
+        {
+            Options.Configure(app);
+            FrameworkOption = app.Option(
+                "-f|--framework",
+                "Target Framework",
+                CommandOptionType.SingleValue);
+
+            ConfigurationOption = app.Option(
+                "-c|--configuration",
+                "Configuration",
+                CommandOptionType.SingleValue);
+
+            NoBuildOption = app.Option(
+                "--no-build",
+                "Do not build project before compiling views.",
+                CommandOptionType.NoValue);
+            app.OnExecute(() => Execute());
+        }
+
+        private int Execute()
+        {
+            ProjectPath = GetProjectPath();
+            Configuration = ConfigurationOption.Value() ?? DotNet.Cli.Utils.Constants.DefaultConfiguration;
+            TargetFramework = GetTargetFramework();
+            RuntimeContext = GetRuntimeContext();
+            OutputPath = GetBuildOutputPath();
+
+            if (!NoBuildOption.HasValue())
+            {
+                var exitCode = BuildProject();
+                if (exitCode != 0)
+                {
+                    return exitCode;
+                }
+            }
+
+            var dispatchArgs = new List<string>
+            {
+                ProjectPath,
+                "--output",
+                OutputPath,
+                "--content-root",
+                Options.ContentRootOption.Value() ?? Directory.GetCurrentDirectory(),
+            };
+
+            if (Options.ConfigureCompilationType.HasValue())
+            {
+                dispatchArgs.Add("--configure-compilation-type");
+                dispatchArgs.Add(Options.ConfigureCompilationType.Value());
+            }
+
+            if (Options.GeneratePdbOption.HasValue())
+            {
+                dispatchArgs.Add("--generate-pdbs");
+            }
+
+            var toolName = typeof(Design.Program).GetTypeInfo().Assembly.GetName().Name;
+            var dispatchCommand = DotnetToolDispatcher.CreateDispatchCommand(
+                dispatchArgs,
+                TargetFramework,
+                Configuration,
+                outputPath: OutputPath,
```

Comment:
Not entirely sure it applies here - we're not using the output path from the xproj. It's something the user optionally passes in.